From a693e5d67abf6f27f5284c86361604babec529c6 Mon Sep 17 00:00:00 2001
From: Daeho Ro <lamanus@techready.ai>
Date: Sun, 19 Jan 2025 07:03:22 +0000
Subject: [PATCH] PS-9641 (8.0) Fix system ssl error

---
 cmake/ssl.cmake | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/cmake/ssl.cmake b/cmake/ssl.cmake
index 58e87e5d05c3..0fab3269d33b 100644
--- a/cmake/ssl.cmake
+++ b/cmake/ssl.cmake
@@ -434,7 +434,9 @@ MACRO (MYSQL_CHECK_SSL)
     # WITH_SSL values (system, openssl11, /path/to/openssl). This will fix
     # building modules which rely on OpenSSL::SSL and OpenSSL::Crypto only with
     # different openssl variants.
-    ADD_LIBRARY(OpenSSL::SSL UNKNOWN IMPORTED)
+    IF(NOT TARGET OpenSSL::SSL)
+      ADD_LIBRARY(OpenSSL::SSL UNKNOWN IMPORTED)
+    ENDIF()
     SET_TARGET_PROPERTIES(OpenSSL::SSL PROPERTIES
       INTERFACE_INCLUDE_DIRECTORIES "${OPENSSL_INCLUDE_DIR}")
     IF(EXISTS "${MY_OPENSSL_LIBRARY}")
@@ -443,7 +445,9 @@ MACRO (MYSQL_CHECK_SSL)
         IMPORTED_LOCATION "${MY_OPENSSL_LIBRARY}")
     ENDIF()
 
-    ADD_LIBRARY(OpenSSL::Crypto UNKNOWN IMPORTED)
+    IF(NOT TARGET OpenSSL::Crypto)
+      ADD_LIBRARY(OpenSSL::Crypto UNKNOWN IMPORTED)
+    ENDIF()
     SET_TARGET_PROPERTIES(OpenSSL::Crypto PROPERTIES
       INTERFACE_INCLUDE_DIRECTORIES "${OPENSSL_INCLUDE_DIR}")
     IF(EXISTS "${MY_CRYPTO_LIBRARY}")
